shark/shark_runner.py

```diff
@@ -154,3 +159,16 @@ def train(self, num_iters=1):
             weights = self.shark_runner.forward(weights + inputs)
 
         return weights
+
+
+class SharkMode:
```

Comment:
Is this the best name ? What is non-SharkMode ? Stock IREE  ?

Reply:
oh well i have no idea - i just thought of shark as an alternate to "torch-mlir mode", which is just tracing+compilation+refbackend. I can rename to whatever you think is best.

Reply:
In the context of this maybe SharkEagerMode ?

Reply:
how about `JawsMode` since hungry <=> eager?

i'm only half joking...

Reply:
We should use JAWS for something